mllib/src/test/scala/org/apache/spark/ml/classification/LinearSVCSuite.scala

```diff
@@ -51,6 +54,12 @@ class LinearSVCSuite extends SparkFunSuite with MLlibTestSparkContext with Defau
     smallBinaryDataset = generateSVMInput(A, Array[Double](B, C), nPoints, 42).toDF()
     smallValidationDataset = generateSVMInput(A, Array[Double](B, C), nPoints, 17).toDF()
     binaryDataset = generateSVMInput(1.0, Array[Double](1.0, 2.0, 3.0, 4.0), 10000, 42).toDF()
+
+    // Dataset for testing SparseVector
+    smallSparseBinaryDataset = generateSVMInput(A, Array[Double](B, C), nPoints, 42, false).toDF()
```

Comment:
Why call generateSVMInput again?  It seems brittle.  I'd prefer to use a UDF to convert the vectors to sparse vectors here.